instaloader/instaloadercontext.py

```diff
@@ -47,15 +47,30 @@ class InstaloaderContext:
     class :class:`Instaloader`.
     """
 
-    def __init__(self, sleep: bool = True, quiet: bool = False, user_agent: Optional[str] = None,
-                 graphql_count_per_slidingwindow: Optional[int] = None, max_connection_attempts: int = 3):
+    def __init__(
+            self,
+            sleep: bool = True,
+            quiet: bool = False,
+            user_agent: Optional[str] = None,
+            graphql_count_per_slidingwindow: Optional[int] = None,
+            max_connection_attempts: int = 3,
+            proxy: Optional[str] = None,
+    ):
 
         self.user_agent = user_agent if user_agent is not None else default_user_agent()
         self._session = self.get_anonymous_session()
         self.username = None
         self.sleep = sleep
         self.quiet = quiet
         self.max_connection_attempts = max_connection_attempts
+        if proxy:
+            if 'https' not in proxy:
```

Comment:
```suggestion
            if not proxy.startswith('https'):
```

because `not in` searchs anywhere in the string